src/lib.rs

```diff
@@ -112,12 +112,29 @@ use std::collections::HashSet;
 use std::path;
 use walrus::ir::VisitorMut;
 
+/// Input configuration.
+#[derive(Clone, Debug)]
+pub enum Input {
+    /// The input `.wasm` file that should have its function snipped.
+    File(path::PathBuf),
+    /// The input WebAssembly blob that should have its function snipped.
+    Buffer(Vec<u8>),
+    // TODO: Support Walrus module directly.
+    // Module(walrus::Module),
+}
+
+impl Default for Input {
+    fn default() -> Self {
+        Input::File(path::PathBuf::default())
```

Comment:
Defaulted here to the original behaviour, but it may as well could just work with an empty buffer.